<a href="https://colab.research.google.com/github/KAlikhanov/Project_2/blob/main/Dataset_Choices_and_Analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [24]:
import pandas as pd

In [25]:
filename = '/content/drive/MyDrive/Colab Notebooks/Project 2/ds_salaries.csv'
df = pd.read_csv(filename)
df.head()

Unnamed: 0  work_year experience_level employment_type  \
0           0       2020               MI              FT   
1           1       2020               SE              FT   
2           2       2020               SE              FT   
3           3       2020               MI              FT   
4           4       2020               SE              FT   

                    job_title  salary salary_currency  salary_in_usd  \
0              Data Scientist   70000             EUR          79833   
1  Machine Learning Scientist  260000             USD         260000   
2           Big Data Engineer   85000             GBP         109024   
3        Product Data Analyst   20000             USD          20000   
4   Machine Learning Engineer  150000             USD         150000   

  employee_residence  remote_ratio company_location company_size  
0                 DE             0               DE            L  
1                 JP             0               JP            S  
2                 GB            50               GB            M  
3                 HN             0               HN            S  
4                 US            50               US            L

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 607 entries, 0 to 606
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Unnamed: 0          607 non-null    int64 
 1   work_year           607 non-null    int64 
 2   experience_level    607 non-null    object
 3   employment_type     607 non-null    object
 4   job_title           607 non-null    object
 5   salary              607 non-null    int64 
 6   salary_currency     607 non-null    object
 7   salary_in_usd       607 non-null    int64 
 8   employee_residence  607 non-null    object
 9   remote_ratio        607 non-null    int64 
 10  company_location    607 non-null    object
 11  company_size        607 non-null    object
dtypes: int64(5), object(7)
memory usage: 57.0+ KB


First choice:
1. [Source.](https://www.kaggle.com/datasets/ruchi798/data-science-job-salaries)
2. Data scientist job salaries and other relevant data.
3. The target is the 'salary_in_usd' column.
4. This is a regression problem as the value we are trying to predict is continuous.
5. There are 10 features, however I plan on dropping some of them, for example the Unnamed: 0 column is useless. Also, the salary and salary_currency columns are redundant.
6. There are 607 rows of data.
7. This dataset is more complicated than my second option which may cause time problems for me in the future. I am also unsure if I should keep the country columns and it is something I will have to decide.

In [27]:
filename2 = '/content/drive/MyDrive/Colab Notebooks/Project 2/collegePlace.csv'
df2 = pd.read_csv(filename2)
df2.head()

Age  Gender                         Stream  Internships  CGPA  Hostel  \
0   22    Male  Electronics And Communication            1     8       1   
1   21  Female               Computer Science            0     7       1   
2   22  Female         Information Technology            1     6       0   
3   21    Male         Information Technology            0     8       0   
4   22    Male                     Mechanical            0     8       1   

   HistoryOfBacklogs  PlacedOrNot  
0                  1            1  
1                  1            1  
2                  0            1  
3                  1            1  
4                  0            1

In [28]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2966 entries, 0 to 2965
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Age                2966 non-null   int64 
 1   Gender             2966 non-null   object
 2   Stream             2966 non-null   object
 3   Internships        2966 non-null   int64 
 4   CGPA               2966 non-null   int64 
 5   Hostel             2966 non-null   int64 
 6   HistoryOfBacklogs  2966 non-null   int64 
 7   PlacedOrNot        2966 non-null   int64 
dtypes: int64(6), object(2)
memory usage: 185.5+ KB


Second choice:
1. [Source.](https://www.kaggle.com/datasets/tejashvi14/engineering-placements-prediction?select=collegePlace.csv)
2. University's placement records for an engineering course. Datasets includes anonymous student backgrounds as well as whether or not they got placed.
3. The 'PlacedOrNot' column is the target.
4. This is a classification problem as we are trying to predict a class.
5. There are 6 features.
6. There are 2966 rows of data.
7. I am unsure what 'Stream' means but the information contained in the column is understandable, I am simply unsure why the column is called Stream.